In [68]:
library(tidyverse)
library(palmerpenguins)
library(broom)
library(modelr)

options(repr.plot.width = 10, repr.plot.height = 5) 

# Linear Regression Model
**Lecture 2 - Demo**

In [6]:
penguins_clean <-
    penguins %>%
    drop_na() %>%
    select(-year)

penguins_clean %>%
    slice_sample(n = 5)

species   island bill_length_mm bill_depth_mm flipper_length_mm body_mass_g
1 Adelie    Dream  41.5           18.5          201               4000       
2 Gentoo    Biscoe 45.0           15.4          220               5050       
3 Gentoo    Biscoe 48.5           14.1          220               5300       
4 Chinstrap Dream  50.3           20.0          197               3300       
5 Chinstrap Dream  45.7           17.0          195               3650       
  sex   
1 male  
2 male  
3 male  
4 male  
5 female

## 1. **One Binary Covariate**

### 1.1 **What the model looks like**

#### 1.1.1 **Dummy Variables**

### 1.2 **Fitting using R**

### 1.3 **Equivalent to t-test from STAT 201**

## 2. **Categorical covariate with more than 2 categories**

### 2.1 **What the model looks like**

#### 2.1.1 **Dummy Variables**

### 2.2 **Fitting using R**

### 2.3 Difference in means?

## 3. One Categorical and One Numerical 

### 3.1 **What the model looks like**

### 3.2 **Fitting using R**

In this model, the difference in the means between the groups do not depend on the `flipper_length_mm`.

## 4. **MLR with interaction**

### 4.1 **What the model looks like**

### 4.2 **Fitting using R**